internal/js/modules/k6/browser/tests/page_test.go

```diff
@@ -1036,6 +1036,74 @@ func TestPageClose(t *testing.T) {
 func TestPageOn(t *testing.T) {
 	t.Parallel()
 
+	tests := map[string]struct {
+		fun     string
+		wantErr string
+	}{
+		"nil on('console') handler": {
+			fun:     `page.on('console')`,
+			wantErr: `TypeError: The "listener" argument must be a function`,
+		},
+		"valid on('console') handler": {
+			fun: `page.on('console', () => {})`,
+		},
+		"nil on('metric') handler": {
+			fun:     `page.on('metric')`,
+			wantErr: `TypeError: The "listener" argument must be a function`,
+		},
+		"valid on('metric') handler": {
+			fun: `page.on('metric', () => {})`,
+		},
+		"nil on('request') handler": {
+			fun:     `page.on('request')`,
+			wantErr: `TypeError: The "listener" argument must be a function`,
+		},
+		"valid on('request') handler": {
+			fun: `page.on('request', () => {})`,
+		},
+		"nil on('response') handler": {
+			fun:     `page.on('response')`,
+			wantErr: `TypeError: The "listener" argument must be a function`,
+		},
+		"valid on('response') handler": {
+			fun: `page.on('response', () => {})`,
+		},
+	}
```

Comment:
Since `page.on` runs for every `page.on` metric event, I believe checking only one event is enough to test it, rather than going through every event name. Otherwise, we'd have to keep maintaining this test for every event type we add. It's not the end of the world, but I don't think it's worth it. No strong opinion!